src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -1676,28 +1676,31 @@ public bool IsScopeBlock(SyntaxNode node)
             => node.IsKind(SyntaxKind.Block);
 
         public bool IsExecutableBlock(SyntaxNode node)
-            => node.IsKind(SyntaxKind.Block, SyntaxKind.SwitchSection);
+            => node.IsKind(SyntaxKind.Block, SyntaxKind.SwitchSection, SyntaxKind.CompilationUnit);
 
-        public SyntaxList<SyntaxNode> GetExecutableBlockStatements(SyntaxNode node)
+        public IReadOnlyList<SyntaxNode> GetExecutableBlockStatements(SyntaxNode node)
```

Comment:
this does worry me in that previously anyone calling this thought they were getting a contiguous sequence of statements.  the global-statement case is def different as clients really need to be able to understand the disctinction here between top-level statements and body-level statements.

Reply:
The sequence is contiguous in valid code.

Reply:
oh?  do we not allow global statements to intermix with non-global-statements in legal code? i thoght that was fine.

Reply:
It's allowed in script, but not top-level statements. Note that while the syntax nodes are contiguous, they are not siblings because each item is wrapped in a `GlobalStatementSyntax`.